lightkurve/lightcurve.py

```diff
@@ -47,16 +47,24 @@ class LightCurve(object):
     time_scale : str
         String which specifies how the time is measured,
         e.g. tdb', 'tt', 'ut1', or 'utc'.
+    targetid : str
+        Identifier of the target.
     meta : dict
         Free-form metadata associated with the LightCurve.
     """
-    def __init__(self, time, flux=None, flux_err=None, time_format=None,
-                 time_scale=None, meta={}):
-        self.time = np.asarray(time)
+    def __init__(self, time=None, flux=None, flux_err=None, time_format=None,
+                 time_scale=None, targetid=None, meta={}):
+        if time is None and flux is None:
+            raise ValueError('either time or flux must be given')
+        if time is None:
+            self.time = np.arange(len(flux))
+        else:
+            self.time = np.asarray(time)
```

Comment:
If we want to be rigorous here, then let's also check that ``len(flux) == len(time)``?

Reply:
Ok!

Reply:
Oh, `_validate_array` already takes care of checking this,